mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -82,9 +83,29 @@ protected void commit() throws MailboxException {
             throw new MailboxException("Commit of transaction failed", e);
         }
     }
+
+    @Override
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        Preconditions.checkArgument(mailbox.getMailboxId() == null, "A mailbox we want to create should not have a mailboxId set already");
+
+        try {
+            if (isPathAlreadyUsedByAnotherMailbox(mailbox)) {
```

Comment:
do we have a transaction opened?
Can't we rely on the database to detect duplicate insertion instead?

Reply:
Moved to an other PR -> #3108, code of create for other implementation is not necessary for just create adoption on Cassandra (which is the primary goal of the ticket)